daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/unparsers/UState.scala

```diff
@@ -422,6 +430,15 @@ final class UStateMain private (
     this(inputter, new VariableBox(vmap), diagnosticsArg, dataProcArg,
       dataOutputStream, initialSuspendedExpressions, tunable)
 
+  /**
+   * Peek at the infoset inputter, and get the root element so we don't lose
+   * access to it.
+   */
+  val originalInfosetElement: DIElement = {
```

Comment:
This  depends on the fact that initialization when creating a new UState pushes the rootERD and then inspects, which caches the root element, allowing us to get it here.